Trimble.Modus.Components/Controls/TMDataGrid/DataGrid.xaml.cs

```diff
@@ -0,0 +1,727 @@
+    {
+        var columnToSort = Columns[sortData.Index];
+
+        IList<object> items;
+
+        switch (sortData.Order)
+        {
+            case SortingOrder.Ascendant:
+                items = unsortedItems.OrderBy(x => ReflectionUtils.GetValueByPath(x, columnToSort.PropertyName)).ToList();
+                _ = columnToSort.SortingIcon.RotateTo(0, 300, easing: Easing.Linear);
+                break;
+            case SortingOrder.Descendant:
+                items = unsortedItems.OrderByDescending(x => ReflectionUtils.GetValueByPath(x, columnToSort.PropertyName)).ToList();
+                _ = columnToSort.SortingIcon.RotateTo(180, 300, easing: Easing.Linear);
+                break;
+            case SortingOrder.None:
+                items = unsortedItems.ToList();
+                break;
+            default:
+                throw new NotImplementedException();
+        }
+
+        foreach (var column in Columns)
+        {
+            if (column == columnToSort)
+            {
+                column.SortingOrder = sortData.Order;
+                column.SortingIconContainer.IsVisible = sortData.Order != SortingOrder.None;
+            }
+            else
+            {
+                column.SortingOrder = SortingOrder.None;
+                column.SortingIconContainer.IsVisible = false;
+            }
+        }
+
+        return items;
+    }
+
+    private void SortColumn(SortData? sortData = null)
+    {
+        if (ItemsSource is null)
+        {
+            return;
+        }
+
+        sortData ??= SortedColumnIndex;
+
+        var originalItems = ItemsSource.Cast<object>().ToList();
+
+        IList<object> sortedItems;
+
+        if (CanSort(sortData))
+        {
+            sortedItems = GetSortedItems(originalItems, sortData);
+        }
+        else
+        {
+            sortedItems = originalItems;
+        }
+
+
+        InternalItems = sortedItems;
+    }
+    #endregion Sorting methods
+
+    #region Methods
+
+    /// <summary>
+    /// Scrolls to the row
+    /// </summary>
+    /// <param name="item">Item to scroll</param>
+    /// <param name="position">Position of the row in screen</param>
+    /// <param name="animated">animated</param>
+    public void ScrollTo(object item, ScrollToPosition position, bool animated = true) => _collectionView.ScrollTo(item, position: position, animate: animated);
+
+    #endregion Methods
+
+    #region Bindable properties
+    public static readonly BindableProperty ReadOnlyListProperty = BindableProperty.Create(nameof(ReadOnlyList), typeof(List<int>), typeof(DataGrid), defaultValueCreator: _ => new List<int>());
+    public static readonly BindableProperty ItemSizingStrategyProperty = BindableProperty.Create(nameof(ItemSizingStrategy), typeof(ItemSizingStrategy), typeof(DataGrid), DeviceInfo.Platform == DevicePlatform.Android ? ItemSizingStrategy.MeasureAllItems : ItemSizingStrategy.MeasureFirstItem);
```

Comment:
Need of ItemSizingStrategyProperty?

Reply:
This property is available in MAUI CollectionView, hence added it here

Reply:
This property is now removed